sqlglot/dialects/snowflake.py

```diff
@@ -167,14 +167,21 @@ def _datatype_sql(self: Snowflake.Generator, expression: exp.DataType) -> str:
     return self.datatype_sql(expression)
 
 
-def _regexpilike_sql(self: Snowflake.Generator, expression: exp.RegexpILike) -> str:
-    flag = expression.text("flag")
+def _regexplike_sql(self: Snowflake.Generator, expression: exp.RegexpILike | exp.RegexpLike) -> str:
+    flag = expression.args.get("flag")
 
-    if "i" not in flag:
-        flag += "i"
+    if isinstance(expression, exp.RegexpILike):
+        case_insensitive_flag = exp.Literal.string("i")
+        flag = self.func("CONCAT", case_insensitive_flag, flag) if flag else case_insensitive_flag
 
+    # Arguments with collation specifications are currently not supported, so we
+    # use an empty string as collation, which indicates to use no collation. See:
+    # https://docs.snowflake.com/en/sql-reference/functions/regexp_like
     return self.func(
-        "REGEXP_LIKE", expression.this, expression.expression, exp.Literal.string(flag)
+        "REGEXP_LIKE",
+        self.func("COLLATE", expression.this, "''"),
```

Comment:
why are you adding collate here?

Reply:
shouldn't you just remove collate and warn if you see it?

Reply:
Discussed in Slack - both having explicit collations here and @ DDL time are user errors and we don't need to do something about them.